devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy

```diff
@@ -0,0 +1,392 @@
+	log.info "MultiChannelEndPointReport called: cmd '${cmd}'"
+	if(!childDevices) {
+		if (isDawonWallSmartSwitch3()) {
+			addChildSwitches(3)
+		} else if (isDawonWallSmartSwitch2()) {
+			addChildSwitches(2)
+		} else if (isDawonWallSmartSwitch1()) {
+			addChildSwitches(1)
+		} else {
+			log.debug "child endpoint=$cmd.endPoints"
+			addChildSwitches(cmd.endPoints)
+		}
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.ManufacturerSpecificReport cmd) {
+	log.info "ManufacturerSpecificReport called: cmd '${cmd}'"
+	log.debug "manufacturerId:   ${cmd.manufacturerId}"
+	log.debug "productId:        ${cmd.productId}"
+	log.debug "productTypeId:    ${cmd.productTypeId}"
+	def msr = String.format("%04X-%04X-%04X", cmd.manufacturerId, cmd.productTypeId, cmd.productId)
+	updateDataValue("MSR", msr)
+	createEvent([descriptionText: "$device.displayName MSR: $msr", isStateChange: false])
+}
+
+private sendCommands(cmds, delay=1000) {
+	log.info "sendCommands called: cmds '${cmds}', delay '${delay}'"
+	sendHubCommand(cmds, delay)
+}
+
+private commands(commands, delay=200) {
+	log.info "commands called: commands '${commands}', delay '${delay}'"
+	delayBetween(commands.collect{ command(it) }, delay)
+}
+
+private secure(cmd) {
+	log.info "secure called"
+	if(zwaveInfo.zw.endsWith("s")) {
+		zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+	} else {
+		cmd.format()
+	}
+}
+
+private encap(cmd, endpoint = null) {
+	log.info "encap called"
+	if (endpoint) {
+		zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint:endpoint).encapsulate(cmd)
+	} else {
+		cmd
+	}
+}
+
+private secureEncap(cmd, endpoint = null) {
+	log.info "secureEncap called"
+	secure(encap(cmd, endpoint))
+}
+
+private changeSwitch(endpoint, value) {
+	log.info "changeSwitch called: vlaue: '${value}', endpoint: '${endpoint}'"
+	def result = []
+	if(endpoint) {
+		String childDni = "${device.deviceNetworkId}:$endpoint"
+		def child = childDevices.find { it.deviceNetworkId == childDni }
+		log.debug "changeSwitch: endpoint '${endpoint}', value: '${value}')"
+		result << child.sendEvent(name: "switch", value: value)
+	}
+	result
+}
+
+def childOnOff(deviceNetworkId, value) {
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) sendHubCommand onOffCmd(value, switchId)
+}
+
+def childRefresh(deviceNetworkId) {
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) sendHubCommand refresh(switchId)
+}
+
+def isDawonWallSmartSwitch3() {
+	return zwaveInfo.prod.equals("0063")
+}
+
+def isDawonWallSmartSwitch2() {
+	return zwaveInfo.prod.equals("0062")
+}
+
+def isDawonWallSmartSwitch1() {
+	return zwaveInfo.prod.equals("0061")
+}
+
+private onOffCmd(value, endpoint) {
+	log.info "onOffCmd called: val:${value}, ep:${endpoint}"
+	delayBetween([
+		secureEncap(zwave.basicV1.basicSet(value: value), endpoint),
+		"delay 500"
+	], 500)
```

Comment:
I'm guessing there's meant to be a get here?

Also, it seems unnecessary to put an explicit delay command here when you're calling `delayBetween`

Reply:
Thanks. I will remove the "delayBetween".